scripts/exit.py

```diff
@@ -4,19 +4,435 @@
+    )
+    serialized_blocks = list(map(serialize_block, checkpoint_blocks))
+
+    burn_tx_serialized_block = serialized_blocks[burn_tx_block_number - block_start]
+    merkle_tree = MerkleTree(serialized_blocks)
+
+    return merkle_tree.get_proof(burn_tx_serialized_block)
+
+
+@hot_swap_network("polygon-main")
+def build_receipt_proof(burn_tx_receipt: TxReceipt, burn_tx_block: BlockData) -> List[bytes]:
+    """Build the burn_tx_receipt proof."""
+    state_sync_tx_hash = keccak256(
+        b"matic-bor-receipt-" + burn_tx_block["number"].to_bytes(8, "big") + burn_tx_block["hash"]
+    )
+    receipts_trie = HexaryTrie({})
+    receipts = (
+        web3.eth.get_transaction_receipt(tx)
+        for tx in burn_tx_block["transactions"]
+        if tx != state_sync_tx_hash
+    )
+    for tx_receipt in receipts:
+        path = rlp.encode(tx_receipt["transactionIndex"])
+        receipts_trie[path] = serialize_receipt(tx_receipt)
+
+    key = rlp.encode(burn_tx_receipt["transactionIndex"])
+    proof = receipts_trie.get_proof(key)
+
+    assert (
+        receipts_trie.root_hash == burn_tx_block["receiptsRoot"]
+    ), "Receipts trie root is incorrect"
+
+    return key, proof
+
+
+def find_log_index(burn_tx_receipt: TxReceipt) -> int:
+    """Retrieve the index of the burn event log."""
+    ERC20_TRANSFER_EVENT_SIG = keccak256(b"Transfer(address,address,uint256)")
+    for idx, log in enumerate(burn_tx_receipt["logs"]):
+        topics = log["topics"]
+        if topics[0] == ERC20_TRANSFER_EVENT_SIG and topics[2] == HexBytes(0) * 32:
+            return idx
+
+    # this should not be reached
+    raise Exception("Transfer Log Event Not Found in Burn Tx Receipt")
+
+
+def encode_payload(
+    header_block_number: int,
+    block_proof: List[bytes],
+    block_number: int,
+    timestamp: int,
+    transactions_root: bytes,
+    receipts_root: bytes,
+    burn_tx_receipt: TxReceipt,
+    receipt_proof: List[bytes],
+    path: bytes,
+    log_index: int,
+) -> bytes:
+    """RLP encode the data required to form the calldata for exiting."""
+    payload = [
+        header_block_number,
+        b"".join(block_proof),
+        block_number,
+        timestamp,
+        transactions_root,
+        receipts_root,
+        serialize_receipt(burn_tx_receipt),
+        rlp.encode(receipt_proof),
+        HexBytes(0) + path,
+        log_index,
+    ]
+    return rlp.encode(payload)
+
+
+def build_calldata(burn_tx_id: str = MATIC_BURN_TX_ID) -> bytes:
+    """Generate the calldata required for withdrawing ERC20 asset on Ethereum."""
+    assert is_burn_checkpointed()
+
+    burn_tx, burn_tx_receipt, burn_tx_block = fetch_burn_tx_data(burn_tx_id)
+    start, end, header_block_number = fetch_block_inclusion_data(burn_tx_block["number"])
+    block_proof = build_block_proof(start, end, burn_tx_block["number"])
+    path, receipt_proof = build_receipt_proof(burn_tx_receipt, burn_tx_block)
+
+    log_index = find_log_index(burn_tx_receipt)
```

Comment:
In the event of an incorrect txid, this function will raise. If we move it to before the proof generation, we can save ourselves a few minutes when there's a mistake.